girder/molecules/molecules/molecule.py

```diff
@@ -463,3 +465,25 @@ def search(self, params):
             .pagingParams(defaultSort='_id',
                           defaultSortDir=SortDir.DESCENDING,
                           defaultLimit=25))
+
+    @access.user
+    @autoDescribeRoute(
+            Description('Generate 3D coordinates for a molecule.')
+            .param('id', 'The id of the molecule', paramType='path')
+            .errorResponse('Molecule not found.', 404)
+    )
+    def generate_3d_coords(self, id):
+        """Generate 3D coords if not present and not being generated"""
+        user = self.getCurrentUser()
+
+        mol = MoleculeModel().load(id, user=user, level=AccessType.WRITE)
```

Comment:
If you use `modelParam` in `autoDescribeRoute` the loading will be done for you.

Reply:
Done.